### PR TITLE
Implement breadcrumbs

### PR DESCRIPTION
**Pull request by Lucianosc**

### CODE REVIEW

**Comment by Corantin** [comment]
Should we have pool number and proposal number instead ?
image

File: apps/web/components/Breadcrumbs.tsx
Comment (Mati0x) [inline apps/web/components/Breadcrumbs.tsx:1]: Suggestion:
For the breadcrumb navigation:
* Use the Pool ID instead of the Pool name when navigating to the pool.
* Apply the same logic to Proposals: use IDs instead of names.

**Review by Mati0x — COMMENTED** [review]
VERY COOL FEATURE

**Comment by Corantin** [comment]
image
Mayeb att an ellipsis

File: apps/web/components/Breadcrumbs.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    })();
+  }, [breadcrumbsPromise]);
+
+  if (!breadcrumbs.length) return null;
+
+  return (
+    <div aria-label="Breadcrumbs" className="flex items-center justify-center">
+      <ol className="flex items-center justify-center">
+        {breadcrumbs.map(({ href, label }, index) => (
+          <li
```
Comment (Corantin) [inline apps/web/components/Breadcrumbs.tsx:60]: Maybe add an ellipsis: 
image
https://gardens-v2-git-implement-breadcrumbs-1hive.vercel.app/gardens/421614/0xcc6c8b9f745db2277f7aac1bc026d5c2ea7bd88d/0x535f7dc4916ee558f8d6600adb067224d4a72254

File: apps/web/components/Breadcrumbs.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            key={href}
+            className="flex items-center justify-center text-neutral-soft-content"
+          >
+            {index !== 0 && <ChevronRightIcon className="mx-[6px] h-5 w-5" />}
```
Comment (Corantin) [inline apps/web/components/Breadcrumbs.tsx:64]:
```suggestion
            {index !== 0 && <ChevronRightIcon className="mx-1 h-5 w-5" />}
```
Reason: https://uxplanet.org/everything-you-should-know-about-8-point-grid-system-in-ux-design-b69cb945b18d
TLDR: tends to make the ui more alligned with components

File: apps/web/components/Breadcrumbs.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+const truncateString = (str: string) => {
+  return `${str.slice(0, 6)}...${str.slice(-4)}`;
+};
+
+export function Breadcrumbs() {
+  const path = usePathname();
+  const [breadcrumbs, setBreadcrumbs] = useState<Breadcrumb[]>([]);
+
+  const breadcrumbsPromise = useMemo(async (): Promise<Breadcrumb[]> => {
```
Comment (Corantin) [inline apps/web/components/Breadcrumbs.tsx:21]: Not sure that useMemo is saving computing in that case, this is only saving the function declaration witch is nothing but the code inside is not executed until called so no need toi memoize it

Comment (Corantin) [reply apps/web/components/Breadcrumbs.tsx:21]: you can go with simple function declaration

Comment (Corantin) [reply apps/web/components/Breadcrumbs.tsx:21]: And using useMemo is costly so we shouldnt use it if it doesnt bring significant performance saving

File: apps/web/components/Breadcrumbs.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      })
+      .filter((segment): segment is Breadcrumb => segment !== undefined);
+  }, [path]);
+
+  useEffect(() => {
+    (async () => {
+      const result = await breadcrumbsPromise;
+      setBreadcrumbs(result);
+    })();
+  }, [breadcrumbsPromise]);
```
Comment (Corantin) [inline apps/web/components/Breadcrumbs.tsx:52]: so this should be path instead

File: apps/web/components/Breadcrumbs.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+  if (!breadcrumbs.length) return null;
```
Comment (Corantin) [inline apps/web/components/Breadcrumbs.tsx:54]: More html friendly maybe ?
```suggestion
  if (!breadcrumbs.length) return <></>;
```

**Review by Corantin — CHANGES_REQUESTED** [review]

File: apps/web/app/(app)/gardens/[chain]/[garden]/[community]/[poolId]/create-proposal/page.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  params: { chain: string; poolId: number; garden: string };
+}) {
+  const { data } = useSubgraphQueryByChain<getPoolDataQuery>(
+    chain,
+    getPoolDataDocument,
+    { poolId: poolId, garden: garden },
+  );
+  const strategyObj = data?.cvstrategies?.[0];
+
+  const [metadata, setMetadata] = useState({ title: "", description: "" });
```
[the remark this answers is not in the document]
Comment (Lucianosc) [reply apps/web/app/(app)/gardens/[chain]/[garden]/[community]/[poolId]/create-proposal/page.tsx:27]: Screenshot from 2024-07-08 20-15-13

File: apps/web/components/Breadcrumbs.tsx (hunk 3 above)
Comment (Lucianosc) [reply apps/web/components/Breadcrumbs.tsx:21]: got it!

File: apps/web/components/Breadcrumbs.tsx (hunk 1 above)
Comment (Lucianosc) [reply apps/web/components/Breadcrumbs.tsx:60]: whats an ellipsis? should i just truncate the string? like max cap it to 30 chars or something and then add "..."?

Comment (kamikazebr) [reply apps/web/components/Breadcrumbs.tsx:60]: That is what ellipsis is. Here some example https://www.npmjs.com/package/react-lines-ellipsis but maybe implement manually for you guys make more sense

Comment (Corantin) [reply apps/web/components/Breadcrumbs.tsx:60]: or just use css

Comment (Corantin) [reply apps/web/components/Breadcrumbs.tsx:60]: text-ellipsis I think in tailwind

File: apps/web/app/(app)/gardens/[chain]/[garden]/[community]/[poolId]/create-proposal/page.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (Corantin) [reply apps/web/app/(app)/gardens/[chain]/[garden]/[community]/[poolId]/create-proposal/page.tsx:27]: this is not related to breadcrumbs, its for the create-proposal page

File: apps/web/components/Breadcrumbs.tsx (hunk 1 above)
Comment (Lucianosc) [reply apps/web/components/Breadcrumbs.tsx]: Screenshot from 2024-07-09 12-12-48
implemented

File: apps/web/app/(app)/gardens/[chain]/[garden]/[community]/[poolId]/create-proposal/page.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (Lucianosc) [reply apps/web/app/(app)/gardens/[chain]/[garden]/[community]/[poolId]/create-proposal/page.tsx:28]: https://github.com/1Hive/gardens-v2/pull/278/commits/8dece1d2a1f2999c8b870bed2efb5a0ee5eb69ce

**Review by Mati0x — COMMENTED** [review]
Like the #pool id and #proposal id

**Review by Corantin — APPROVED** [review]

**Comment by Corantin** [comment]
Added a fix from useSubgraphQuery signature changed

**Comment by Corantin** [comment]
Good job looking great 🔥
